From 9f25c6c5d3122c92e864e6cba296cbbdbfbb756d Mon Sep 17 00:00:00 2001
From: Sakari Ailus <sakari.ailus@linux.intel.com>
Date: Mon, 20 Jun 2016 15:42:56 +0300
Subject: [PATCH 6/9] v4l: subdev: Add source route flag

Add a flag to tell the route is a a source of the stream.
This flag is only for the intermediate solution,
so need to be remove.

Change-Id: I8718a687ef7e5b24797b7cf80364dd17e3a3f1a9
Tracked-On:
Signed-off-by: Sakari Ailus <sakari.ailus@linux.intel.com>
Reviewed-on: https://android.intel.com/519866
Tested-by: Tao, Wei <wei.tao@intel.com>
Reviewed-by: Zheng, Jian Xu <jian.xu.zheng@intel.com>
Reviewed-by: Ding, Yunliang <yunliang.ding@intel.com>
Reviewed-by: Yang, Fan F <fan.f.yang@intel.com>
Reviewed-by: Meng, WeiX <weix.meng@intel.com>
Tested-by: Meng, WeiX <weix.meng@intel.com>
---
 include/uapi/linux/v4l2-subdev.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/uapi/linux/v4l2-subdev.h b/include/uapi/linux/v4l2-subdev.h
index 5e29237..6d750ff 100644
--- a/include/uapi/linux/v4l2-subdev.h
+++ b/include/uapi/linux/v4l2-subdev.h
@@ -166,6 +166,7 @@ struct v4l2_subdev_selection {
 
 #define V4L2_SUBDEV_ROUTE_FL_ACTIVE	(1 << 0)
 #define V4L2_SUBDEV_ROUTE_FL_IMMUTABLE	(1 << 1)
+#define V4L2_SUBDEV_ROUTE_FL_SOURCE	(1 << 2)
 
 /**
  * struct v4l2_subdev_route - A signal route inside a subdev
@@ -181,6 +182,10 @@ struct v4l2_subdev_selection {
  *
  *	V4L2_SUBDEV_ROUTE_FL_IMMUTABLE: Is the stream immutable, i.e.
  *	can it be activated and inactivated? Set by the driver.
+ *
+ *	V4L2_SUBDEV_ROUTE_FL_SOURCE: Is the sub-device the source of a
+ *	stream? In this case the sink information is unused (and
+ *	zero). Set by the driver.
  */
 struct v4l2_subdev_route {
 	__u32 sink_pad;
-- 
2.9.0

